src/vm/coreassemblyspec.cpp

```diff
@@ -170,6 +170,9 @@ VOID  AssemblySpec::Bind(AppDomain      *pAppDomain,
         hr = CreateAssemblyNameObject(&pName, assemblyDisplayName, CANOF_PARSE_DISPLAY_NAME, NULL);
         if (SUCCEEDED(hr))
         {
+            Holder<Thread*, DoNothing, Thread::ResetParentOfLoadAssemblyByName> holder(GetThread());
+            if(GetThread() != NULL)
+            	GetThread()->SetParentOfLoadAssemblyByName(GetParentAssembly());
```

Comment:
Storing these things on thread object looks pretty scary. Can these things be nested on the same thread? Is it really guaranteed that things won't get lost or otherwise messed up in the tunnel?